JPS_BASE_LIB/src/test/java/uk/ac/cam/cares/jps/base/converter/test/TBoxGenerationTest.java

```diff
@@ -140,7 +148,93 @@ public void testClassEquivalentRelation(){
+	}
+
+	/**
+	 * This is to test if the Domain-Range that are Union-ed will not be created when:
+	 * (a) there is a presence of a Quantifier in the Domain-Range relationship
+	 * AND 
+	 * (b) when there IS-A relation between the ancestors of the Domain and Range with the same Object Property  
+	 */
+	@Test
+	public void testSkipDomainRangeCreationUnion(){
+		OWLOntology generatedOntology = generateOntology(SAMPLE_CSV_FILE_PATH+SAMPLE_TBOX_CSV_TEMPLATE_FILE_7);
+		if (generatedOntology == null) {
+			throw new JPSRuntimeException("The requested CSV file could not be converted into an ontology.");
+		}
+
+		OWLOntology referenceOntology = readReferenceOntology(REFERENCE_OWL_FILE_PATH+REFERENCE_OWL_FILE_7);
+		if (referenceOntology == null) {
+			throw new JPSRuntimeException("The requested reference ontology could not be read from the provided path.");
+		}
+		
+		assertEquals(referenceOntology.getAxiomCount(), generatedOntology.getAxiomCount());
+		assertEquals(referenceOntology.getAxioms(), generatedOntology.getAxioms());		
+	}
+
+	/**
+	 * This is to test if the Domain-Range that are Intersect-ed will not be created when:
+	 * (a) there is a presence of a Quantifier in the Domain-Range relationship
+	 * AND 
+	 * (b) when there IS-A relation between the ancestors of the Domain and Range with the same Object Property  
+	 */
+	@Test
+	public void testSkipDomainRangeCreationIntersection(){
+		OWLOntology generatedOntology = generateOntology(SAMPLE_CSV_FILE_PATH+SAMPLE_TBOX_CSV_TEMPLATE_FILE_8);
+		if (generatedOntology == null) {
+			throw new JPSRuntimeException("The requested CSV file could not be converted into an ontology.");
+		}
+
+		OWLOntology referenceOntology = readReferenceOntology(REFERENCE_OWL_FILE_PATH+REFERENCE_OWL_FILE_8);
+		if (referenceOntology == null) {
+			throw new JPSRuntimeException("The requested reference ontology could not be read from the provided path.");
+		}
+		
+		assertEquals(referenceOntology.getAxiomCount(), generatedOntology.getAxiomCount());
+		assertEquals(referenceOntology.getAxioms(), generatedOntology.getAxioms());		
+	}
 	
+	/**
+	 * This is to test if the Domain-Range creation with a mix of Union and Intersect are created correctly
+	 */
+	@Test
+	public void testSkipDomainRangeCreationWithUnionAndIntersectionMix(){
+		OWLOntology generatedOntology = generateOntology(SAMPLE_CSV_FILE_PATH+SAMPLE_TBOX_CSV_TEMPLATE_FILE_9);
+		if (generatedOntology == null) {
+			throw new JPSRuntimeException("The requested CSV file could not be converted into an ontology.");
+		}
+
+		OWLOntology referenceOntology = readReferenceOntology(REFERENCE_OWL_FILE_PATH+REFERENCE_OWL_FILE_9);
+		if (referenceOntology == null) {
+			throw new JPSRuntimeException("The requested reference ontology could not be read from the provided path.");
+		}
+		
+		assertEquals(referenceOntology.getAxiomCount(), generatedOntology.getAxiomCount());
+		assertEquals(referenceOntology.getAxioms(), generatedOntology.getAxioms());		
+	}
+
+
```

Comment:
Add the following method here and create Unit tests for all domains and ranges, including complex ones in files 2-9.

	/**
	 * Tests whether the domain of the hasSibling relation is Person.
	 */
	@Test
	public void testDomainOfHasSiblingProperty() {
		String objectProperty = "<http://www.theworldavatar.com/ontology/ontokin/OntoKin.owl#hasSibling>";
		String domain = "<http://xmlns.com/foaf/0.1/Person>";
		String axiomType = "ObjectPropertyDomain";
		OWLOntology generatedOntology = generateOntology(SAMPLE_CSV_FILE_PATH + SAMPLE_TBOX_CSV_TEMPLATE_FILE_3);
		for (OWLAxiom owlAxiom : generatedOntology.getAxioms()) {
			if (owlAxiom.getNNF().getAxiomType().toString().equalsIgnoreCase(axiomType)) {
				for (OWLObjectProperty objectPropertySignature : owlAxiom.getObjectPropertiesInSignature()) {
					if(objectPropertySignature.toString().equalsIgnoreCase(objectProperty)) {
						assertTrue(owlAxiom.toString().contains(domain));
						assertEquals(axiomType+"("+objectProperty+" "+domain+")",  owlAxiom.toString());
				}
				}
			}
		}
	}